http-api/src/lib.rs

```diff
@@ -406,6 +511,92 @@ async fn root_handler() -> Result<impl Reply, Rejection> {
     Ok(warp::reply::json(&response))
 }
 
+/// Return the peer id for the node identity.
+/// `GET /v1/peer`
+async fn peer_handler(peer_id: PeerId) -> Result<impl warp::Reply, warp::Rejection> {
+    let response = json!({
+        "id": peer_id.to_string(),
+    });
+    Ok(warp::reply::json(&response))
+}
+
+/// `GET /v1/sessions/:session-id`
+async fn session_get_handler(
+    ctx: Context,
+    id: String,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    let sessions = ctx.sessions.read().await;
+    let session = sessions.get(&id).ok_or(Error::NotFound)?;
+
+    match session {
+        AuthState::Authorized(session) => {
+            Ok(warp::reply::json(&json!({ "id": id, "session": session })))
+        }
+        AuthState::Unauthorized {
+            nonce,
+            expiration_time,
+        } => Ok(warp::reply::json(
+            &json!({ "id": id, "nonce": nonce, "expiration_time": expiration_time }),
+        )),
+    }
+}
+
+/// `POST /v1/sessions`
+async fn session_create_handler(ctx: Context) -> Result<impl warp::Reply, warp::Rejection> {
+    let expiration_time =
+        Utc::now() + chrono::Duration::from_std(UNAUTHORIZED_SESSIONS_EXPIRATION).unwrap();
+    let mut sessions = ctx.sessions.write().await;
+    let (session_id, nonce) = create_session(&mut sessions, expiration_time);
+
+    let response = json!({ "id": session_id, "nonce": nonce });
+
+    Ok(warp::reply::json(&response))
+}
+
+/// `PUT /v1/sessions/:session_id`
+async fn session_signin_handler(
+    ctx: Context,
+    id: String,
+    request: AuthRequest,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    // Get unauthenticated session data, return early if not found
+    let mut sessions = ctx.sessions.write().await;
+    let session = sessions.get(&id).ok_or(Error::NotFound)?;
+
+    if let AuthState::Unauthorized { nonce, .. } = session {
+        let message = Message::from_str(request.message.as_str())
+            .map_err(|_| Error::Auth("Not able to parse message"))?;
```

Comment:
Is it a siwe::ParseError returned here?

Reply:
Yes and we should be able to return a `Error::SiweParseError`
Good catch